index.js

```diff
@@ -179,19 +180,24 @@ class ftdi extends EventEmitter {
 
   async closeAsync() {
     this.isClosing = true;
-    await this.device.releaseInterface(0);
-    await this.device.close();
+    try {
+      console.log('Sending EOT');
+      await this.writeAsync([0x04]);
+      await delay(2000); // wait for send/receive to complete
+      await this.device.releaseInterface(0);
+      await this.device.close();
+      this.removeAllListeners();
+      console.log('Closed device');
+    } catch(err) {
+      console.log('Error:', err);
+    }
   }
 
   close(cb) {
-    this.isClosing = true;
     (async () => {
-      await this.device.releaseInterface(0);
-      await this.device.close();
+      await this.closeAsync();
       return cb();
-    })().catch((error) => {
-      return cb(error);
```

Comment:
Was it the intention to remove `cb(error)`?

Reply:
Yep. Since it's `close()` is only a callback wrapper for `async closeAsync()`, which only prints out the error and doesn't throw it anymore, it seemed unnecessary.